src/v/cluster/tx_gateway_frontend.cc

```diff
@@ -1480,46 +1480,49 @@ tx_gateway_frontend::do_commit_tm_tx(
             }
         }
 
-        std::vector<ss::future<prepare_tx_reply>> pfs;
-        for (auto rm : tx.partitions) {
-            pfs.push_back(_rm_partition_frontend.local().prepare_tx(
-              rm.ntp,
-              rm.etag,
-              model::tx_manager_ntp.tp.partition,
-              tx.pid,
-              tx.tx_seq,
-              timeout));
-        }
-
         std::vector<ss::future<prepare_group_tx_reply>> pgfs;
         for (auto group : tx.groups) {
             pgfs.push_back(_rm_group_proxy->prepare_group_tx(
               group.group_id, group.etag, tx.pid, tx.tx_seq, timeout));
         }
 
-        if (
-          !is_transaction_ga()
-          && tx.status == tm_transaction::tx_status::ongoing) {
-            auto preparing_tx = co_await stm->mark_tx_preparing(
-              expected_term, tx.id);
-            if (!preparing_tx.has_value()) {
-                if (preparing_tx.error() == tm_stm::op_status::not_leader) {
-                    outcome->set_value(tx_errc::not_coordinator);
-                    co_return tx_errc::not_coordinator;
+        std::vector<ss::future<prepare_tx_reply>> pfs;
+        pfs.reserve(tx.partitions.size());
+        auto ok = true;
+        auto rejected = false;
+
+        if (!is_transaction_ga()) {
+            for (auto rm : tx.partitions) {
+                pfs.push_back(_rm_partition_frontend.local().prepare_tx(
+                  rm.ntp,
+                  rm.etag,
+                  model::tx_manager_ntp.tp.partition,
+                  tx.pid,
+                  tx.tx_seq,
+                  timeout));
+            }
+
+            if (tx.status == tm_transaction::tx_status::ongoing) {
+                auto preparing_tx = co_await stm->mark_tx_preparing(
+                  expected_term, tx.id);
+                if (!preparing_tx.has_value()) {
+                    if (preparing_tx.error() == tm_stm::op_status::not_leader) {
+                        outcome->set_value(tx_errc::not_coordinator);
+                        co_return tx_errc::not_coordinator;
```

Comment:
is there a ticket tracking the ignored exceptional future issue with `pfs`?

Reply:
https://github.com/redpanda-data/redpanda/issues/6875